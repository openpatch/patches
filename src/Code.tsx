/** @jsx jsx */
import { ReactNode } from "react";
import { jsx } from "@emotion/react";
import {
  Prism as SyntaxHighlighter,
  SyntaxHighlighterProps,
} from "react-syntax-highlighter";
import { atomDark } from "react-syntax-highlighter/dist/cjs/styles/prism";

export type CodeProps = {
  children: ReactNode;
} & SyntaxHighlighterProps;

export const Code = ({ children, ...props }: CodeProps) => {
  return (
    <SyntaxHighlighter style={atomDark} {...props}>
      {children}
    </SyntaxHighlighter>
  );
};
