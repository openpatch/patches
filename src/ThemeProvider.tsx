import React, { ReactElement } from "react";
import { ThemeProvider as EThemeProvider } from "emotion-theming";
import { Theme, base } from "./themes";
import { GlobalCssTheme } from "./themes/utils";

interface ThemeProviderProps {
  theme?: Theme;
  children: ReactElement;
}

const ThemeProvider = ({ theme = base, children }: ThemeProviderProps) => {
  return (
    <EThemeProvider theme={theme}>
      <GlobalCssTheme theme={theme} />
      {children}
    </EThemeProvider>
  );
};

export default ThemeProvider;
