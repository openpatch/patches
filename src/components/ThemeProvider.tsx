import React, { ReactElement } from "react";
import {
  ThemeProvider as SCThemeProvider,
  DefaultTheme,
} from "styled-components";
import defaultTheme from "../shared/theme";

interface ThemeProviderProps {
  theme?: DefaultTheme;
  children: ReactElement;
}

const ThemeProvider = ({
  theme = defaultTheme,
  children,
}: ThemeProviderProps) => {
  return <SCThemeProvider theme={theme}>{children}</SCThemeProvider>;
};

export default ThemeProvider;
