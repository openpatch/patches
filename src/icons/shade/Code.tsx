import * as React from "react";

function CodeIcon(props: React.SVGProps<SVGSVGElement>) {
  return (
    <svg className="icon-code" viewBox="0 0 24 24" {...props}>
      <rect width={18} height={18} x={3} y={3} className="primary" rx={2} />
      <path
        d="M8.7 13.3a1 1 0 01-1.4 1.4l-2-2a1 1 0 010-1.4l2-2a1 1 0 111.4 1.4L7.42 12l1.3 1.3zm6.6 0l1.29-1.3-1.3-1.3a1 1 0 111.42-1.4l2 2a1 1 0 010 1.4l-2 2a1 1 0 01-1.42-1.4zm-3.32 3.9a1 1 0 01-1.96-.4l2-10a1 1 0 011.96.4l-2 10z"
        className="secondary"
      />
    </svg>
  );
}

export default CodeIcon;
