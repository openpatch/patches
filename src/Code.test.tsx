import { render } from "@testing-library/react";
import { Code, CodeProps } from "./Code";
import { ThemeProvider } from "./ThemeProvider";

const renderWithTheme = (props?: CodeProps) =>
  render(
    <ThemeProvider>
      <Code {...props}>public static void;</Code>
    </ThemeProvider>
  );

test("should render", () => {
  renderWithTheme();
});

test("should match snapshot", () => {
  const { container } = renderWithTheme();
  expect(container).toMatchSnapshot();
});
