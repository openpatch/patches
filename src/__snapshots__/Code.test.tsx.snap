// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <div
    id="patches-root"
  >
    <pre
      style="color: rgb(197, 200, 198); text-shadow: 0 1px rgba(0, 0, 0, 0.3); font-family: Inconsolata, Monaco, Consolas, 'Courier New', Courier, monospace; direction: ltr; text-align: left; white-space: pre; word-spacing: normal; word-break: normal; line-height: 1.5; tab-size: 4; hyphens: none; padding: 1em; margin: .5em 0px; overflow: auto; border-radius: 0.3em; background: rgb(29, 31, 33);"
    >
      <code
        style="color: rgb(197, 200, 198); text-shadow: 0 1px rgba(0, 0, 0, 0.3); font-family: Inconsolata, Monaco, Consolas, 'Courier New', Courier, monospace; direction: ltr; text-align: left; white-space: pre; word-spacing: normal; word-break: normal; line-height: 1.5; tab-size: 4; hyphens: none;"
      >
        <span>
          public static void;
        </span>
      </code>
    </pre>
  </div>
</div>
`;
